### ISSUE TITLE
two questions about the proposed method

### ISSUE THREAD

**Issue by Yiluan** [hujiecpp/InformationCompetingProcess#2]
Hi Jie,

Thanks for sharing the code. I have two questions about your paper.

1) In the classification experiments, it seems like you are training two feature extractors to output y and z for ICP. In Table 1, however, VIB, DIM*, VIBx2, DIMx2, ICP-ALL are all training one feature extractor while ICP-COM maintains two. Am I right? If this is true, is the comparison between methods with two feature extractors *fair* with methods with only one?

2) The discussion in Sec. 4.4 is intuitively correct. However, letting the feature extractor maximize the reconstruction loss is equivalent to minimizing the logP_(z|y) (or logP_(y|z)), which is the lower bound of MI. Usually minimizing the lower bound can hardly minimize the true value. Could you pls tell me where I am wrong? 

Thank you again for your hard work!

**Comment by hujiecpp** [comment]
Thanks for your interest. The answers are listed below:

1. Not correct. For rigorous experimental settings, VIB, DIM* were trained by using one feature extractor but VIBx2, DIMx2, ICP-ALL, ICP-COM were all trained by the same setting as ICP (two feature extractors but different types of constraints).

2. Sec.4.4 aims to minimize the MI between z and y, and actually this is equal to letting z and y be independent with each other. From the perspective of Predictability Minimization [1], we introduce one extra predictor H to achieve this goal, which is not equal to minimizing logP_(z|y)(or logP_(y|z)) but MI(z, y).

[1] Schmidhuber J. Learning factorial codes by predictability minimization[J]. Neural Computation, 1992, 4(6): 863-879.

**Comment by Yiluan** [comment]
Thanks for your quick reply.

1) I thought VIBx2, DIMx2 only expended "the representation dimension of both methods to the same size of ICP's" (Sec. 5.1.2). It's interesting to see that both VIBx2 and DIMx2 perform worse than their simper versions. 
I am most curious about the baseline setup in my previous question since there is only one feature extractor for it. Have you conducted any experiments to see the performance of simple ensemble of baseline models? For example, average of two baseline modles that are initialized and trained separately.  

2) I will take a look at the paper. Thank you.

**Comment by hujiecpp** [comment]
Thank you for the great suggestion. I think the reason VIBx2 and DIMx2 perform worse than their simper versions is that the model's capacity is too large and overfits the training set when convergence.
Averaging two baseline models that are initialized and trained separately is like calculating the error bars for the baseline model from different initialization, and I chose the best result among them when the training convergence.

**Comment by Yiluan** [comment]
Thank you for your answers! Help a lot!
